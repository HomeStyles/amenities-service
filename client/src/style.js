import styled from 'styled-components';

export const UppercaseText = styled.div`
  font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif;
  font-size: 12px;
  font-style: normal;
  font-weight: 700;
  letter-spacing: 0.2px;
  line-height: 16px;
  text-transform: uppercase;
  color: #a02f18;
`;

export const Header = styled.div`
  font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif;
  font-size: 17px;
  font-weight: 300;
  letter-spacing: 0.2px;
  line-height: 22px;
  text-transform: none;
  color: #484848;
`;

export const Text = Header.extend`
  font-family: Circular,"Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 14px;
  line-height: 1.43;
`;

export const BlueLink = Text.extend`
  font-weight: 400;
  color: #008489;
`;

export const Box = styled.div`
  float: left;
  padding: 5px 10px 5px 10px;
  margin-left: 5px;
  margin-top: 10px;
  margin-bottom: 30px;
  margin-right: 1000px;
  width: 300px;
  border: 1px solid #c4c4c4;
  border-radius: 2px;
`;

export const Stats = Box.extend`
  border: none;
  margin-bottom: 10px;
`;

export const RoundPhoto = styled.div`
  margin-left: 10px;
  margin-top: 10px;
  margin-bottom: 30px;
  width: 50%;
  height: 50%;
  border-radius: 50%;
`;