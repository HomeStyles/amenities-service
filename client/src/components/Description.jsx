import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';
import axios from 'axios';
import Modal from 'react-modal';
import {UppercaseText, Header, RoundPhoto, Text, BlueLink, Box, Stats, GrayDivider} from '../style.js';

Modal.setAppElement('body');

class Description extends React.Component {
  constructor() {
    super();

    this.state = {
      homeId: 1,
      home: {},
      amenities: {},
      rooms: {},
      host: {},
      modal1IsOpen: false,
      modal2IsOpen: false,
    };
    this.openModal1 = this.openModal1.bind(this);
    this.openModal2 = this.openModal2.bind(this);
    this.closeModal1 = this.closeModal1.bind(this);
    this.closeModal2 = this.closeModal2.bind(this);
    this.createHome = this.createHome.bind(this);
  }

  fetchHomes(id) {
    axios.get(`http://127.0.0.1:3002/amenities/${id}`)
      .then((response) => {
        this.setState({
          home: response.data,
          amenities: response.data.amenities,
          rooms: response.data.rooms,
          host: response.data.host,
        });
        console.log('state set correctly', this.state.host);
      })
      .catch((err) => {
        throw err;
      });
  }
  
  createHome() {
    axios.post('http://127.0.0.1:3002/amenities/form', {
      home: {},
      amenities: {},
      rooms: {},
      host: {},
    })
      .then((response) => {
        console.log('this is the response', response);
      })
      .catch((error) => {
        throw err;
      });
  }

  openModal1() {
    this.setState({modal1IsOpen: true});
  }

  openModal2() {
    this.setState({modal2IsOpen: true});
  }

  closeModal1() {
    this.setState({modal1IsOpen: false});
  }

  closeModal2() {
    this.setState({modal2IsOpen: false});
  }

  componentDidMount() {
    let app = this;
    this.fetchHomes(app.state.homeId);
  }

  render() {
    return (
      <div>
        <UppercaseText>ENTIRE APARTMENT</UppercaseText>
        <Header><h2>{this.state.home.homeName}</h2></Header>
        <RoundPhoto><img style={{borderRadius: '50%'}} src={this.state.host.image}/></RoundPhoto>
        <Text>{this.state.home.location}<br></br>
        </Text>
        <Stats>
          <Text>
            {this.state.rooms.totalBedrooms} bedrooms, {this.state.rooms.totalBeds} beds, {this.state.rooms.totalBaths} baths
          </Text>
        </Stats>
        <Box>
          <Text><strong>This home is on people’s minds.</strong><br></br>It's been viewed {this.state.home.viewsThisWeek}+ times in the past week.
          </Text>
        </Box>
        <Text>
          {this.state.home.description}<br></br>
        </Text>
        <br></br>
        <BlueLink>Read more about the space ^</BlueLink>
        <br></br>
        <br></br>
        <BlueLink onClick={this.openModal1}>
          Contact host
        </BlueLink>
        <GrayDivider></GrayDivider>
        <Modal isOpen={this.state.modal1IsOpen}>
          <button onClick={this.closeModal1}>close</button>
          <Header><h2>Contact {this.state.host.firstName}</h2></Header>
          <Text>
          Once you send a message, {this.state.host.firstName} can invite you to book their home.<br></br><br></br>
            Make sure you share the following:<br></br><br></br>
            <li>Tell {this.state.host.firstName} a little about yourself</li>
            <li>What brings you to {this.state.home.location}? Who’s joining you?</li>
            <li>What do you love about this listing? Mention it!</li>
          </Text>
        </Modal>
        <Text>
          <h4>Amenities</h4>
          Kitchen, wifi, iron
        </Text>
        <br></br>
        <BlueLink onClick={this.openModal2}>
          Show all 7 amenities
        </BlueLink>
        <GrayDivider></GrayDivider>
        <Modal isOpen={this.state.modal2IsOpen}>
          <button onClick={this.closeModal2}>close</button>
          <Header><h2>Amenities</h2></Header>
          <Text>
            <strong>Basic</strong>
            <ul>Wifi</ul>
            <ul>Iron</ul>
            <ul>Washer</ul>
            <ul>Air conditioning</ul>
            <strong>Facilities</strong><br></br>
            <strong>Dining</strong><br></br>
            <strong>Guest access</strong><br></br>
            <strong>Not included</strong><br></br>
          </Text>
        </Modal>
        <Text>
          <h4>Sleeping arrangements</h4>
          <Box>1 double bed, 1 sofa bed</Box>
        </Text>
        <GrayDivider></GrayDivider>
        <Text>
          <h4>House Rules</h4>Not suitable for pets<br></br>No parties or events<br></br>Check-in is anytime after {this.state.home.checkIn}<br></br>Check out by {this.state.home.checkOut}
        </Text>
        <br></br>
        <BlueLink>
          Read all Rules ^
        </BlueLink>
        <GrayDivider></GrayDivider>
      </div>
    );
  }
}

export default Description;