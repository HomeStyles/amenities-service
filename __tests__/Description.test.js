import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount, render} from 'enzyme';
import Description from '../client/src/components/Description.jsx';
import Modal from 'react-modal';
import {UppercaseText, Header, RoundPhoto, Text, BlueLink, Box, Stats} from '../client/src/style.js';
import 'jest-styled-components';

describe('app rendered', () => { 
  it('Description should render without crashing', () => {
    expect(shallow(<Description />)
    );
  });
});

it('renders amenities header', () => {
  const wrapper = shallow(<Description />);
  const amenities = <h4>Amenities</h4>;
  expect(wrapper.contains(amenities)).toEqual(true);
});

test('links render in blue', () => {
  const link = renderer.create(<BlueLink />).toJSON();
  expect(link).toMatchSnapshot();
  expect(link).toHaveStyleRule('color', '#008489');
});

describe('<Modal /> component', () => {
  it('should change state when modal is closed', () => {
    const wrapper = shallow(<Description />);
    const modal1 = wrapper.find('button').at(0);
    const modal2 = wrapper.find('button').at(1);

    modal1.simulate('click');
    expect(wrapper.state('modal1IsOpen')).toEqual(false);

    modal2.simulate('click');
    expect(wrapper.state('modal1IsOpen')).toEqual(false);
  });
  it('should render a modal when blue link is clicked', () => {
    const openModal1 = jest.fn();
    const wrapper = shallow(<BlueLink onClick={openModal1}/>);
    expect(wrapper).toMatchSnapshot();
    wrapper.simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});