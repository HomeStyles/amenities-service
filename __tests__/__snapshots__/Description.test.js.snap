// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> component should render a modal when blue link is clicked 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bxivhb caWaAu",
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bxivhb caWaAu",
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Modal /> component should render a modal when blue link is clicked 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bxivhb caWaAu",
      "onClick": [MockFunction] {
        "calls": Array [
          Array [],
        ],
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bxivhb caWaAu",
        "onClick": [MockFunction] {
          "calls": Array [
            Array [],
          ],
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`links render in blue 1`] = `
.c0 {
  font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif;
  font-size: 17px;
  font-weight: 300;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  line-height: 22px;
  text-transform: none;
  color: #484848;
  font-family: Circular,"Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 14px;
  line-height: 1.43;
  font-weight: 400;
  color: #008489;
}

<div
  className="c0"
/>
`;
